src/libostree/ostree-repo.c

```diff
@@ -4691,7 +4687,7 @@ _ostree_repo_gpg_verify_with_metadata (OstreeRepo          *self,
 
       if (!_ostree_gpg_verifier_add_keyring_dir (verifier, self->repodir,
                                                  cancellable, error))
-        goto out;
+        return NULL;
```

Comment:
Yes, I was wondering how would you react to it. I changed it here, because the `out` label wasn't doing any cleanups at all.